lib/membrane_h264_plugin/parser/au_timestamp_generator.ex

```diff
@@ -17,30 +17,33 @@ defmodule Membrane.H264.Parser.AUTimestampGenerator do
     %{
       au_counter: 0,
       key_frame_au_idx: 0,
-      prev_pic_vcl_nalu: nil,
+      prev_pic_first_vcl_nalu: nil,
       prev_pic_order_cnt_msb: 0
     }
   end
 
   @spec generate_ts_with_constant_framerate(
           [NALu.t()],
-          framerate :: {frames :: pos_integer(), seconds :: pos_integer()},
-          max_frame_reorder :: non_neg_integer(),
+          config :: %{
+            :framerate => {frames :: pos_integer(), seconds :: pos_integer()},
+            optional(:add_dts_offset) => boolean()
+          },
           t
         ) :: {{pts :: non_neg_integer(), dts :: non_neg_integer()}, t}
-  def generate_ts_with_constant_framerate(au, {frames, seconds}, max_frame_reorder, state) do
+  def generate_ts_with_constant_framerate(au, %{framerate: {frames, seconds}} = config, state) do
     %{au_counter: au_counter, key_frame_au_idx: key_frame_au_idx} = state
     first_vcl_nalu = Enum.find(au, &NALuTypes.is_vcl_nalu_type(&1.type))
     {poc, state} = calculate_poc(first_vcl_nalu, state)
     key_frame_au_idx = if poc == 0, do: au_counter, else: key_frame_au_idx
     pts = div((key_frame_au_idx + poc) * seconds * Membrane.Time.second(), frames)
+    max_frame_reorder = if Map.get(config, :add_dts_offset, true), do: 15, else: 0
```

Comment:
let's parametrize that `@15`. I would also consider "parsing" that `add_dts_offset` option in `handle_init` (by which I mean that we can set its default value there)